apps/web/src/components/Nft/index.tsx

```diff
@@ -4,28 +4,28 @@ import { PAGEVIEW } from '@hey/data/tracking';
 import { Card, GridItemEight, GridLayout } from '@hey/ui';
 import { Leafwatch } from '@lib/leafwatch';
 import type { NextPage } from 'next';
-import { useRouter } from 'next/router';
+import { useSearchParams } from 'next/navigation';
+import Custom500 from 'src/app/500';
 import useOpenseaNft from 'src/hooks/opensea/useOpenseaNft';
-import Custom500 from 'src/pages/500';
 import { useEffectOnce } from 'usehooks-ts';
 
 import NftDetails from './Details';
 import NftPageShimmer from './Shimmer';
 
 const ViewNft: NextPage = () => {
-  const {
-    query: { chain, address, token },
-    isReady
-  } = useRouter();
+  const searchParams = useSearchParams();
+  const chain = searchParams.get('chain');
+  const address = searchParams.get('address');
+  const token = searchParams.get('token');
+
   const {
     data: nft,
     loading,
     error
   } = useOpenseaNft({
     chain: parseInt(chain as string),
     address: address as string,
-    token: token as string,
-    enabled: isReady
+    token: token as string
   });
```

Comment:
so in app router, router is always ready?

Reply:
Yeah, I looked the ReturnType Interface and it doesn't exist nor there are any equivlanets


<img width="629" alt="SCR-20231121-thuq" src="https://github.com/heyxyz/hey/assets/62795688/a7bd6af4-b326-4aaf-8058-0ade81ba8af9">